### ISSUE TITLE
Does not work, hangs forever

### ISSUE THREAD

**Issue by itsyoboieltr** [Exact-Realty/ts-multipart-parser#11]
I run this code, and nothing is ever logged. What could be wrong here?

```ts
import { parseMultipartMessage } from '@exact-realty/multipart-parser';

const decoder = new TextDecoder('utf-8');
const encoder = new TextEncoder();

function wait(milliseconds) {
  return new Promise((resolve) => setTimeout(resolve, milliseconds));
}

const stream = new ReadableStream({
  async start(controller) {
    await wait(100);
    controller.enqueue(
      encoder.encode(`--myboundary
Content-Type: text/plain
Content-Length: 3

he`)
    );
    await wait(100);
    controller.enqueue(encoder.encode(`y`));
    controller.close();
  },
});

const boundary = '--myboundary';

for await (const part of parseMultipartMessage(stream, boundary)) {
  console.log(part.headers.get('content-type'));
  console.log(decoder.decode(part.body));
}
```

**Comment by corrideat** [comment]
Hi @itsyoboieltr,

Thanks for reporting this issue! There are a few things that are wrong with your example (maybe that's intentional), chiefly among them the boundaries being wrong.

This is a revised example that works:

```js
import { parseMultipartMessage } from '@exact-realty/multipart-parser';

const decoder = new TextDecoder('utf-8');
const encoder = new TextEncoder();

function wait(milliseconds) {
  return new Promise((resolve) => setTimeout(resolve, milliseconds));
}

const stream = new ReadableStream({
  async start(controller) {
    await wait(100);
    controller.enqueue(
      encoder.encode(`----myboundary\r\nContent-Type: text/plain\r\nContent-Length: 3\r\n\r\nhe`)
    );
    await wait(100);
    controller.enqueue(encoder.encode(`y`));
    await wait(100);
    controller.enqueue(encoder.encode(`\r\n----myboundary--\r\n`));
    controller.close();
  },
});

const boundary = '--myboundary';

for await (const part of parseMultipartMessage(stream, boundary)) {
  console.log(part.headers.get('content-type'));
  console.log(decoder.decode(new Uint8Array(part.body)));
}
```

That said, I agree that this is a bug because your example shouldn't hang forever after the end of a stream, and it should have thrown an exception instead. I'll take a look at this and  release a new version with a fix soon.

Thanks!

**Comment by corrideat** [comment]
Hi again @itsyoboieltr,

This is fixed in version 1.0.13, and it should not hang with your example (it will, however, throw an error because it can't find the boundary).
